rclcpp/src/rclcpp/executors/static_executor_entities_collector.cpp

```diff
@@ -323,8 +323,7 @@ StaticExecutorEntitiesCollector::add_callback_group(
     throw std::runtime_error("Callback group was already added to executor.");
   }
   if (is_new_node) {
-    rclcpp::node_interfaces::NodeBaseInterface::WeakPtr node_weak_ptr(node_ptr);
-    weak_nodes_to_guard_conditions_[node_weak_ptr] = node_ptr->get_notify_guard_condition();
+    weak_nodes_to_guard_conditions_[node_ptr] = node_ptr->get_notify_guard_condition();
```

Comment:
this looks like a false positive, right?
the change looks fine though

Reply:
Yes, seemed like a false positive for me too.